### PR TITLE
Remove deposits from the PoolCollection contract and extend its interface

### PR DESCRIPTION
**Pull request by lbeder**

### CODE REVIEW

File: deploy/tests/000036-upgrade-pool-migrator-v2.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    before(async () => {
+        ({ deployer } = await getNamedAccounts());
+    });
+
+    beforeEach(async () => {
+        poolMigrator = await DeployedContracts.PoolMigratorV1.deployed();
+    });
+
+    it('should  upgrade and configure the pool migrator contract', async () => {
```
Comment (yudilevi) [inline deploy/tests/000036-upgrade-pool-migrator-v2.ts:20]: extra space

File: deploy/scripts/000037-pool-collection-type1-v3.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { deploy, DeployedContracts, execute, InstanceName, setDeploymentMetadata } from '../../utils/Deploy';
+import { NATIVE_TOKEN_ADDRESS } from '../../utils/TokenData';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+const func: DeployFunction = async ({ getNamedAccounts }: HardhatRuntimeEnvironment) => {
+    const { deployer } = await getNamedAccounts();
+
+    const network = await DeployedContracts.BancorNetworkV2.deployed();
```
[the remark this answers is not in the document]
Comment (lbeder) [reply deploy/scripts/000037-pool-collection-type1-v3.ts:9]: Yep. It's easy to make this mistake when the interface and address are the same, but it's still better to be explicit in these cases.

File: deploy/scripts/000037-pool-collection-type1-v3.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const bnt = await DeployedContracts.BNT.deployed();
+    const networkSettings = await DeployedContracts.NetworkSettingsV1.deployed();
```
[the remark this answers is not in the document]
Comment (lbeder) [reply deploy/scripts/000037-pool-collection-type1-v3.ts:11]: 👍

File: deploy/scripts/000037-pool-collection-type1-v3.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const masterVault = await DeployedContracts.MasterVault.deployed();
+
+    const bntPool = await DeployedContracts.BNTPool.deployed();
+    const externalProtectionVault = await DeployedContracts.ExternalProtectionVault.deployed();
+    const poolTokenFactory = await DeployedContracts.PoolTokenFactory.deployed();
+    const poolMigrator = await DeployedContracts.PoolMigratorV1.deployed();
```
[the remark this answers is not in the document]
Comment (lbeder) [reply deploy/scripts/000037-pool-collection-type1-v3.ts:17]: 👍

**Review by yudilevi — APPROVED** [review]

File: deploy/scripts/000039-upgrade-network-info-v2.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { DeployedContracts, InstanceName, setDeploymentMetadata, upgradeProxy } from '../../utils/Deploy';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+const func: DeployFunction = async ({ getNamedAccounts }: HardhatRuntimeEnvironment) => {
+    const { deployer } = await getNamedAccounts();
+
+    const network = await DeployedContracts.BancorNetworkV1.deployed();
```
Comment (yudilevi) [inline deploy/scripts/000039-upgrade-network-info-v2.ts:8]: thees should use the latest versions

File: test/helpers/Factory.ts [hunk 6]
```diff
@@ -468,7 +468,13 @@ const setupPool = async (
         await createPool(reserveToken, network, networkSettings, poolCollection);
 
         await networkSettings.setFundingLimit(reserveToken.address, MAX_UINT256);
-        await poolCollection.requestFundingT(formatBytes32String(''), reserveToken.address, spec.requestedLiquidity);
+        if (spec.requestedLiquidity) {
```
Comment (yudilevi) [inline test/helpers/Factory.ts:471]: unrelated to this PR: should be `requestedFunding`

File: deploy/scripts/000039-upgrade-network-info-v2.ts (hunk 5 above)
Comment (lbeder) [reply deploy/scripts/000039-upgrade-network-info-v2.ts:8]: Yeah. Deleting older deployments would simplify this...

File: test/helpers/Factory.ts (hunk 6 above)
Comment (lbeder) [reply test/helpers/Factory.ts:471]: 👍

**Review by yudilevi — DISMISSED** [review]

File: deploy/scripts/000039-upgrade-network-info-v2.ts (hunk 5 above)
Comment (yudilevi) [reply deploy/scripts/000039-upgrade-network-info-v2.ts]: 👍

File: contracts/pools/interfaces/IPoolCollection.sol [hunk 7]
```diff
@@ -25,7 +25,6 @@ struct Pool {
     bool tradingEnabled; // whether trading is enabled
     bool depositingEnabled; // whether depositing is enabled
     AverageRate averageRate; // the recent average rate
-    uint256 depositLimit; // the deposit limit
```
[the remark this answers is not in the document]
Comment (lbeder) [reply contracts/pools/interfaces/IPoolCollection.sol:28]: `PoolCollection` isn't a proxy, so all the relevant data is explicitly migrated in any case

**Review by yudilevi — DISMISSED** [review]

**Pull request closed without merging**